folsom/src/main/java/com/spotify/folsom/client/ascii/SetRequest.java

```diff
@@ -148,6 +148,12 @@ public void handle(final AsciiResponse response, final HostAndPort server) throw
       case NOT_STORED:
         succeed(MemcacheStatus.ITEM_NOT_STORED);
         return;
+      case VALUE_TOO_LARGE:
+        succeed(MemcacheStatus.VALUE_TOO_LARGE);
+        return;
+      case OUT_OF_MEMORY:
+        succeed(MemcacheStatus.OUT_OF_MEMORY);
+        return;
```

Comment:
Should these succeed or fail?
Binary client succeeds in the corresponding status, so this would be consistent with that.
But semantically aren't these failed operations? Also in the current version of the library these fail, this will be changing that behaviour.

Reply:
I guess returning a completed future with the `VALUE_TOO_LARGE` status would also be consistent with [this check](https://github.com/spotify/folsom/blob/master/folsom/src/main/java/com/spotify/folsom/client/DefaultRawMemcacheClient.java#L227-L230) so it makes sense in that regard as well.